geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -72,4 +52,446 @@ impl GeyserPluginManager {
+
+        Ok(name)
+    }
+
+    pub(crate) fn unload_plugin(&mut self, name: &str) -> JsonRpcResult<()> {
+        // Check if any plugin names match this one
+        let Some(idx) = self.plugins.iter().position(|plugin| plugin.name().eq(name)) else {
+            // If we don't find one return an error
+            return Err(
+                jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: String::from("plugin requested to unload is not loaded"),
+                    data: None,
+                }
+            )
+        };
+
+        // Unload and drop plugin and lib
+        let mut current_plugin = self.plugins.remove(idx);
+        let current_lib = self.libs.remove(idx);
+        current_plugin.on_unload();
+        drop(current_plugin);
+        drop(current_lib);
+
+        Ok(())
+    }
+
+    /// Checks for a plugin with a given `name`.
+    /// If it exists, first unload it.
+    /// Then, attempt to load a new plugin
+    pub(crate) fn reload_plugin(&mut self, name: &str, config_file: &str) -> JsonRpcResult<()> {
+        // Check if any plugin names match this one
+        let Some(idx) = self.plugins.iter().position(|plugin| plugin.name().eq(name)) else {
+            // If we don't find one return an error
+            return Err(
+                jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: String::from("plugin requested to reload is not loaded"),
+                    data: None,
+                }
+            )
+        };
+
+        // Unload and drop current plugin first in case plugin requires exclusive access to resource,
+        // such as a particular port or database.
+        let mut current_plugin = self.plugins.remove(idx);
+        let current_lib = self.libs.remove(idx);
+        current_plugin.on_unload();
+        drop(current_plugin);
+        drop(current_lib);
+
+        // Try to load plugin, library
+        // SAFETY: It is up to the validator to ensure this is a valid plugin library.
+        let (mut new_plugin, new_lib, new_parsed_config_file) =
+            load_plugin_from_config(config_file.as_ref()).map_err(|err| jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: err.to_string(),
+                data: None,
+            })?;
+
+        // Attempt to on_load with new plugin
+        match new_plugin.on_load(&new_parsed_config_file) {
+            // On success, push plugin and library
+            Ok(()) => {
+                self.plugins.push(new_plugin);
+                self.libs.push(new_lib);
+            }
+
+            // On failure, return error
+            Err(err) => {
+                return Err(jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: format!(
+                        "failed to start new plugin (previous plugin was dropped!): {err}"
```

Comment:
Nits: "Failed to start...."

Reply:
fixed this one yesterday, forgot to mark as resolved